mopidy/core/library.py

```diff
@@ -196,14 +210,23 @@ def refresh(self, uri=None):
         :param uri: directory or track URI
         :type uri: string
         """
-        if uri is not None:
-            backend = self._get_backend(uri)
-            if backend:
-                backend.library.refresh(uri).get()
-        else:
-            futures = [b.library.refresh(uri)
-                       for b in self.backends.with_library.values()]
-            pykka.get_all(futures)
+        futures = {}
+        backends = {}
+        uri_scheme = urlparse.urlparse(uri).scheme if uri else None
+
+        for backend_scheme, backend in self.backends.with_playlists.items():
+            backends.setdefault(backend, set()).add(backend_scheme)
+
+        for backend, backend_schemes in backends.items():
+            if uri_scheme is None or uri_scheme in backend_schemes:
+                futures[backend] = backend.library.refresh(uri)
```

Comment:
Yes, this was to only call refresh once per backend for cases with mutiple schemes.